airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/worker_run/TemporalWorkerRunFactory.java

```diff
@@ -48,11 +50,15 @@ public WorkerRun create(final Job job) {
     final UUID connectionId = UUID.fromString(job.getScope());
     return switch (job.getConfigType()) {
       case SYNC -> () -> {
-        final TemporalResponse<StandardSyncOutput> output = temporalClient.submitSync(
-            job.getId(),
-            attemptId,
-            job.getConfig().getSync(),
-            connectionId);
+        final FeatureFlags featureFlags = new EnvVariableFeatureFlags();
```

Comment:
A `FeatureFlags` should be dependency injected into `TemporalWorkerRunFactory` so we don't have configuration interacting with the env directly that we can't control in tests.

Reply:
Done